docs/integrations/httpx.md

```diff
@@ -63,6 +63,152 @@ Let's see a minimal example below. You can run it with `python main.py`:
     asyncio.run(main())
     ```
 
+## Hooks
+
+The `logfire.instrument_httpx()` method also accepts the following hooks:
+
+- `request_hook`: called before sending a request.
+- `response_hook`: called after receiving a response.
+- `async_request_hook`: called before sending an async request.
+- `async_response_hook`: called after receiving an async response.
+
+You can use these hooks to add custom attributes that are not automatically captured by **Logfire**.
+
+### (Async) Request Hook
+
+The `request_hook` and `async_request_hook` hooks are called before sending a request. They receive two arguments:
+
+```py title="main.py" hl_lines="24-27"
+import asyncio
+from collections import defaultdict
+
+import httpx
+from opentelemetry.instrumentation.httpx import RequestInfo
+from opentelemetry.trace import Span
+
+import logfire
+
+
+def set_request_headers(span: Span, request: RequestInfo):
+    method, url, headers, stream, extensions = request
+    attributes: defaultdict[str, str] = defaultdict(list)
+    for key, value in headers.items():
+        key = key.lower()
+        attributes[f'http.request.header.{key}'].append(value)
+    span.set_attributes(attributes)
+
+
+def request_hook(span: Span, request: RequestInfo):
+    """This is called when using a `httpx.Client`."""
+    set_request_headers(span, request)
+
+
+async def async_request_hook(span: Span, request: RequestInfo):
+    """This is called when using a `httpx.AsyncClient`."""
+    set_request_headers(span, request)
+
+
+logfire.configure()
+logfire.instrument_httpx(request_hook=request_hook, async_request_hook=async_request_hook)
+
+url = 'https://httpbin.org/get'
+
+with httpx.Client() as client:
+    client.get(url)
+
+
+async def main():
+    async with httpx.AsyncClient() as client:
+        await client.get(url)
+
+
+asyncio.run(main())
+```
+
+`RequestInfo` is a [`NamedTuple`][typing.NamedTuple] with the following fields:
+
+- `method`: The HTTP method of the request.
+- `url`: The URL of the request.
+- `headers`: The headers of the request.
+- `stream`: The stream of the request.
+- `extensions`: The extensions of the request.
```

Comment:
```suggestion
`RequestInfo` is a [`NamedTuple`][typing.NamedTuple] with the following attributes: `method`, `url`, `headers`, `stream`, and `extensions`.
```

Maybe I'm just fussy, but if we don't have more useful descriptions of these fields then I don't like the redundancy.